src/components/App.js

```diff
@@ -655,3 +655,20 @@ export default class App extends Component {
 }
 
 App.propTypes = {}
+
+const mapStateToProps = (reduxState) => {
+  return reduxState.chat;
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    addNewMessage: ({ key, fromUsername, maybeSenderId, message }) =>
+      dispatch(addMessage({ key, fromUsername, maybeSenderId, message })),
+    addNewUserStatus: ({ fromUsername, userStatus, created }) =>
+      dispatch(addUserStatus(({ fromUsername, userStatus, created }))),
```

Comment:
That seems correct, i have not yet tested this code. Pushed it in case i was not on the same machine.